specs/ScreenCaptureRequested

```diff
@@ -0,0 +1,394 @@
+        // to be a no-op.
+    }
+};
+
+
+
+void PutHandled(CoreWebView2ScreenCaptureEventArgs args)
+{
+    // In the case of an iframe requesting permission to use Screen Capture, the default behavior is
+    // to first raise the ScreenCaptureRequested event off of the CoreWebView2Frame
+    // and invoke it's handlers, and then raise the event off the CoreWebView2
+    // and invoke it's handlers. However, If we set Handled to true on the
+    // CoreWebView2Frame event handler, then we will not raise the
+    // ScreenCaptureRequested event off the CoreWebView2.
+    // 
+    // NotImplementedException could be thrown if underlying runtime did not
+    // implement Handled. However, we only run this code after checking if
+    // CoreWebView2Frame.ScreenCaptureRequested exists, and both exist together,
+    // so it would not be a problem.
+    args.Handled = true;
+}
+```
+
+# API Details
+## C++
+```
+interface ICoreWebView2_20;
+interface ICoreWebView2ScreenCaptureRequestedEventArgs;
+interface ICoreWebView2ScreenCaptureRequestedEventHandler;
+
+interface ICoreWebView2Frame3;
+interface ICoreWebView2FrameScreenCaptureRequestedEventHandler;
+interface ICoreWebView2FrameScreenCaptureRequestedEventArgs;
+
+/// This interface is an extension of `ICoreWebView2` that supports the ScreenCaptureRequested event. 
+// MSOWNERS: stevenwei@microsoft.com 
+[uuid(accc0e97-fa2d-4a8d-ad61-cc9ae57a1825), object, pointer_default(unique)] 
+interface ICoreWebView2_20 : IUnknown { 
+  /// Add an event handler for the `ScreenCaptureRequested` event. 
+  /// `ScreenCaptureRequested` event is raised when the Screen Capture API is requested by the user using getDisplayMedia(). 
+  HRESULT add_ScreenCaptureRequested( 
+      [in] ICoreWebView2ScreenCaptureRequestedEventHandler* eventHandler, 
+      [out] EventRegistrationToken* token); 
+  /// Remove an event handler previously added with `add_ScreenCaptureRequested`. 
+  // MSOWNERS: stevenwei@microsoft.com 
+  HRESULT remove_ScreenCaptureRequested( 
+      [in] EventRegistrationToken token); 
+}   
+/// Receives `ScreenCaptureRequested` events. 
+// MSOWNERS: stevenwei@microsoft.com 
+[uuid(9b5bbea1-4a58-4567-8b42-8781d3986cb4), object, pointer_default(unique)] 
+interface ICoreWebView2ScreenCaptureRequestedEventHandler : IUnknown { 
+  /// Called to provide the event args when a screen capture is requested on 
+  /// a WebView element. 
+  HRESULT Invoke( 
+      [in] ICoreWebView2* sender, 
+      [in] ICoreWebView2ScreenCaptureRequestedEventArgs* args); 
+} 
+/// Event args for the `ScreenCaptureRequested` event. 
+// MSOWNERS: stevenwei@microsoft.com 
+[uuid(a1d309ee-c03f-11eb-8529-0242ac130003), object, pointer_default(unique)] 
+interface ICoreWebView2ScreenCaptureRequestedEventArgs : IUnknown { 
+  /// The associated frame information that requests the screen capture  
+  /// permission. This can be used to grab the frame source, name, frameId,  
```

Comment:
This also includes URI of the frame? I expect knowing the URI will be important to deciding to allow or disallow screen capture

Reply:
Yeah, FrameInfo.Source is the URI of the frame, which also is valid for the top level document.